src/main/java/tools/jackson/databind/ObjectReader.java

```diff
@@ -692,6 +692,61 @@ public ObjectReader withHandler(DeserializationProblemHandler h) {
         return _with(_config.withHandler(h));
     }
 
+    /**
+     * Enables error collection mode by registering a
+     * {@link tools.jackson.databind.deser.CollectingProblemHandler} with default
+     * error limit (100 problems).
+     *
+     * <p>The returned reader is immutable and thread-safe. Each call to
+     * {@link #readValueCollecting} allocates a fresh problem bucket, so concurrent
+     * calls do not interfere.
+     *
+     * <p>Usage:
+     * <pre>
+     * ObjectReader reader = mapper.reader()
+     *     .forType(MyBean.class)
+     *     .collectErrors();
+     *
+     * MyBean bean = reader.readValueCollecting(json);
+     * </pre>
+     *
+     * @return A new ObjectReader configured for error collection
+     * @since 3.1
+     */
+    public ObjectReader collectErrors() {
```

Comment:
I don't really like the names of the new APIs. I don't especially like 'collect' but I think 'Error' is plain wrong.
'Error' has a very explicit meaning in Java - examples include OutOfMemoryError. Exception is more correct than Error here.

Reply:
Agreed.

Reply:
Something like `problemCollectingReader()` etc would make more sense